ext/mysqlnd/mysqlnd_auth.c

```diff
@@ -1021,7 +1185,7 @@ mysqlnd_caching_sha2_get_and_use_key(MYSQLND_CONN_DATA *conn,
 		const char * const passwd,
 		const size_t passwd_len)
 {
-	static RSA *server_public_key;
+	static mysqlnd_rsa_t server_public_key;
```

Comment:
Please combine both lines to there is a initialization. Another question is why this is a static variable? Why a singleton here? I see it was introduced recently by Nikita. For example in mysqlnd_sha256_public_encrypt() the key variable is not declared as static.

Reply:
This definitely shouldn't be static.

FYI this is not my code, I'm just the person running damage control on it. (And oh boy was there much damage to control ... this code is a true piece of art, I've never seen so many distinct bugs crammed into so little code.)

Reply:
ACK, that `static` is a bad idea.